dgraphtest/acl_cluster.go

```diff
@@ -0,0 +1,398 @@
+			"name": username,
+		},
+	}
+	_, err := hc.RunGraphqlQuery(params, true)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (hc *HTTPClient) CreateUser(username, password string) error {
+	addUser := `
+	mutation addUser($name: String!, $pass: String!) {
+		addUser(input: [{name: $name, password: $pass}]) {
+			user {
+				name
+			}
+		}
+	}`
+
+	params := GraphQLParams{
+		Query: addUser,
+		Variables: map[string]interface{}{
+			"name": username,
+			"pass": password,
+		},
+	}
+	_, err := hc.RunGraphqlQuery(params, true)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (hc *HTTPClient) CreateGroup(name string) error {
+	addGroup := `
+	mutation addGroup($name: String!) {
+		addGroup(input: [{name: $name}]) {
+			group {
+				name
+			}
+		}
+	}`
+
+	params := GraphQLParams{
+		Query: addGroup,
+		Variables: map[string]interface{}{
+			"name": name,
+		},
+	}
+	_, err := hc.RunGraphqlQuery(params, true)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (hc *HTTPClient) AddRulesToGroup(group string, rules []AclGrpRules) error {
+	addRuleToGroup := `mutation updateGroup($name: String!, $rules: [RuleRef!]!) {
+		updateGroup(input: {
+			filter: {
+				name: {
+					eq: $name
+				}
+			},
+			set: {
+				rules: $rules
+			}
+		}) {
+			group {
+				name
+				rules {
+					predicate
+					permission
+				}
+			}
+		}
+	}`
+
+	params := GraphQLParams{
+		Query: addRuleToGroup,
+		Variables: map[string]interface{}{
+			"name":  group,
+			"rules": rules,
+		},
+	}
+	_, err := hc.RunGraphqlQuery(params, true)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (hc *HTTPClient) AddToGroup(userName, group string) error {
+	addUserToGroup := `mutation updateUser($name: String!, $group: String!) {
+		updateUser(input: {
+			filter: {
+				name: {
+					eq: $name
+				}
+			},
+			set: {
+				groups: [
+					{ name: $group }
+				]
+			}
+		}) {
+			user {
+				name
+				groups {
+					name
+				}
+			}
+		}
+	}`
+
+	params := GraphQLParams{
+		Query: addUserToGroup,
+		Variables: map[string]interface{}{
+			"name":  userName,
+			"group": group,
+		},
+	}
+	_, err := hc.RunGraphqlQuery(params, true)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (hc *HTTPClient) RemoveUserFromGroup(userName, groupName string) error {
+	removeUserGroups := `mutation updateUser($name: String!, $groupName: String!) {
+			updateUser(input: {
+				filter: {
+					name: {
+						eq: $name
+					}
+				},
+				remove: {
+					groups: [{ name: $groupName }]
+				}
+			}) {
+				user {
+					name
+					groups {
+						name
+					}
+				}
+			}
+		}`
+
+	params := GraphQLParams{
+		Query: removeUserGroups,
+		Variables: map[string]interface{}{
+			"name":      userName,
+			"groupName": groupName,
+		},
+	}
+
+	_, err := hc.RunGraphqlQuery(params, true)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (hc *HTTPClient) RemoveRuleFromGroup(group string, rulePredicate string) error {
```

Comment:
Wouldn't a better name be RemovePredicateFromGroup?

Reply:
and rename rulePredicate -> predicate?

Reply:
we are removing rule from group not predicate